public/components/notebooks/components/main.tsx

```diff
@@ -95,19 +95,32 @@ export class Main extends React.Component<MainProps, MainState> {
 
   // Fetches path and id for all stored notebooks
   fetchNotebooks = () => {
-    return Promise.all([
-      this.props.http.get(`${NOTEBOOKS_API_PREFIX}/savedNotebook/`),
-      this.props.http.get(`${NOTEBOOKS_API_PREFIX}/`),
-    ])
-      .then(([savedNotebooksResponse, secondResponse]) => {
-        const combinedData = {
-          data: [...savedNotebooksResponse.data, ...secondResponse.data],
-        };
-        this.setState(combinedData);
-      })
-      .catch((err) => {
-        console.error('Issue in fetching the notebooks', err.body.message);
-      });
+    if (this.props.dataSourceEnabled) {
+      // If `MDS` is enabled, only fetch from the first endpoint.
+      return this.props.http
+        .get(`${NOTEBOOKS_API_PREFIX}/savedNotebook/`)
+        .then((savedNotebooksResponse) => {
+          this.setState({ data: savedNotebooksResponse.data });
+        })
+        .catch((err) => {
+          console.error('Issue in fetching the notebooks', err.body.message);
+        });
+    } else {
+      // If `MDS` is not enabled, fetch from both endpoints and combine the data.
+      return Promise.all([
+        this.props.http.get(`${NOTEBOOKS_API_PREFIX}/savedNotebook/`),
+        this.props.http.get(`${NOTEBOOKS_API_PREFIX}/`),
```

Comment:
it's not very clear what the difference is, maybe add a comment to say `/savedNotebook/` API returns notebooks stored as saved objects, and the other one returns notebooks stored as observability objects?